src/Worker/ExpirePosts.php

```diff
@@ -46,54 +45,54 @@ public static function execute()
 		}
 
 		if (!empty($expire_days)) {
-			do {
-				Logger::notice('Start deleting expired threads', ['expiry_days' => $expire_days]);
-				$ret = DBA::e("DELETE FROM `item-uri` WHERE `id` IN
-					(SELECT `uri-id` FROM `post-thread` WHERE `received` < UTC_TIMESTAMP() - INTERVAL ? DAY
-						AND NOT `uri-id` IN (SELECT `uri-id` FROM `post-thread-user`
-							WHERE (`mention` OR `starred` OR `wall` OR `pinned`) AND `uri-id` = `post-thread`.`uri-id`)
-						AND NOT `uri-id` IN (SELECT `uri-id` FROM `post-category`
-							WHERE `uri-id` = `post-thread`.`uri-id`)
-						AND NOT `uri-id` IN (SELECT `uri-id` FROM `post-media`
-							WHERE `uri-id` = `post-thread`.`uri-id`)
-						AND NOT `uri-id` IN (SELECT `parent-uri-id` FROM `post-user` INNER JOIN `contact` ON `contact`.`id` = `contact-id` AND `notify_new_posts`
-							WHERE `parent-uri-id` = `post-thread`.`uri-id`)
-						AND NOT `uri-id` IN (SELECT `parent-uri-id` FROM `post-user`
-							WHERE (`origin` OR `event-id` != 0 OR `post-type` = ?) AND `parent-uri-id` = `post-thread`.`uri-id`)
-						AND NOT `uri-id` IN (SELECT `uri-id` FROM `post-content`
-							WHERE `resource-id` != 0 AND `uri-id` = `post-thread`.`uri-id`))
-					ORDER BY `id` LIMIT ?", $expire_days, Item::PT_PERSONAL_NOTE, $limit);
```

Comment:
Is `$limit` still used in this method?

Reply:
Only in a way to control whether an expiry should be done at all.